networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,297 @@
+    where $A$ is the adjacency matrix and $d_i$ is the degree of node $i$.
+    Note that the definition presented here is valid for unweighted, weighted,
+    directed, and undirected networks, so in the more general case,
+    a non-zero entry of the adjacency matrix $A_{ij}=w_{ij}$
+    represents the existence of a link from node $i$ to node $j$
+    with a weight $w_{ij}$. In general, one will notice that important
+    nodes identified by DomiRank will be connected to a lot of other
+    unimportant nodes. However, other positionally important nodes
+    can also be dominanted by joint-dominance of nodes, that work together
+    in order to dominate another positionally important node. This centrality
+    has a lot of interesting emergent phenomena, so it is recommend to
+    see [1] for more information. That being said, DomiRank can also be
+    expressed in its analytical form, where, the competition can now be
+    supercharged - i.e. $\sigma \in [0,\inf]$. The analytical equation
+    takes the form:
+
+    .. math::
+
+        \boldsymbol{\Gamma} = \sigma (\sigma A + I_{N\times N})^{-1}  A  \boldsymbol{1}_{N\times 1},
+
+    where $I$ is the identity matrix. This analytical equation can be solved
+    as a linear system.
+
+    Interestingly, DomiRank tends to have only positive values for
+    relatively low competition levels (small $\sigma$), however, as
+    the competition levels increase, negative values might emerge.
+    Nodes with negative dominance represent completely submissive nodes,
+    which instead of fighting for their resources/dominance, directly
+    give up these resources to their neighborhood.
+
+    Finally, it is not required that the network is weakly or strongly
+    connected, and can be applied to networks with many components.
+
+    Parameters
+    ----------
+    G : graph
+      A networkx graph.
+
+    analytical: bool, optinal (default=False)
+      Boolean representing if the analytical or iterative formulation
+      for computing DomiRank should be used.
+
+    sigma: float, optional (default=0.95)
+     The level of competition for DomiRank.
+
+    dt: float, optional (default=0.1)
+     The step size for the Newton iteration.
+
+    epsilon: float, optional (default=1e-5)
+     The relative stopping criterion for convergence.
+
+    max_iter: integer, optional (default=50)
+      Maximum number of Newton iterations allowed.
+
+    check_step: integer, optional (default=10)
+     The number of steps between checking whether or
+     not DomiRank has converged/diverged.
+     Note, this paramter must satisfy check_step > 2.
+
+
+    Returns
+    -------
+    nodes : dictionary
+       Dictionary of nodes with DomiRank centrality as the value.
+
+    sigma : float
+       The input $\sigma \in [0,1]$ normalized by the smallest eigenvalue
+
+    converged: boolean
+       A boolean representing whether or not the method did not diverge.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(4)
```

Comment:
```suggestion
    >>> G = nx.path_graph(5)
```
The expected output you give is for `nx.path_graph(5)`.